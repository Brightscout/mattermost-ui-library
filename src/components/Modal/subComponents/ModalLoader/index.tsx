import React from 'react';

import LinearLoader from '@Components/Loader/Linear';

type ModalLoaderProps = {
    loading?: boolean;
    className?: string,
}

const ModalLoader = ({loading, className = ''}: ModalLoaderProps): JSX.Element => (
    <div className={`modal__loader-container position-relative ${className}`}>
        {loading && <LinearLoader/>}
    </div>
);

export default ModalLoader;
