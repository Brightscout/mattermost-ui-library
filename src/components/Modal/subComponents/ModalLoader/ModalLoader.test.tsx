import React from 'react';
import {mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';

import ModalLoader from '.';
import LinearLoader from '@Components/Loader/Linear';

// Mock props data
const ModalLoaderClassName = 'modal-loader-class';

describe('Modal Body', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ModalLoader className={ModalLoaderClassName} />);
    })

    afterEach(() => {
        component.unmount();
    })

    it('Should render correctly and matches snapshot when loading is false', () => {
        expect(component).toMatchSnapshot();
    })

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(ModalLoaderClassName)).toBeTruthy();
    })

    describe('Modal Loader - loading state', () => {
        let shallowRenderedComponent: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

        beforeEach(() => {
            shallowRenderedComponent = shallow(<ModalLoader loading />)
        })

        it('Should render correctly and matches snapshot when loading is true', () => {
            expect(shallowRenderedComponent).toMatchSnapshot();
        })

        it('Should render the linear loader when loading is true', () => {
            expect(shallowRenderedComponent.contains(<LinearLoader />)).toBeTruthy();
        })
    })
})
