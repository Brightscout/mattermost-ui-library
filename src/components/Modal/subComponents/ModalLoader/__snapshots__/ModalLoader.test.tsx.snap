// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Body Modal Loader - loading state Should render correctly and matches snapshot when loading is true 1`] = `ShallowWrapper {}`;

exports[`Modal Body Should render correctly and matches snapshot when loading is false 1`] = `ReactWrapper {}`;
